src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -89,6 +94,39 @@ def refresh_repo(self):
         self.invoke_package_manager(self.cmd_clean_cache)
         self.invoke_package_manager(self.cmd_repo_refresh)
 
+    # region Fetch SnapshotTime
+    def __get_posix_time_from_health_store_id(self, execution_config):
+        """Converts date str received (in format: yyyy.mm.dd) to POSIX time string"""
+        # eg: Input: 2024.12.20 (str) -> Output: 1734681600 (str)
+        posix_time = str()
+        health_store_id = execution_config.health_store_id
+        date_in_health_store_id = execution_config.validate_date_in_health_store_id(health_store_id)
```

Comment:
Breaks separation of concerns. I'd get what's needed here from what is already persisted in ExecutionConfig as maxpatchpublishdate.

Because maxpatchpublishdate can be set in other ways and include more than just a date with no time. 

The package manager does not need to know anything beyond the fully qualified timestamp (including HHMMSS) - the fact that they are 00000 in default patching is incidental and not to be defaulted to.

Reply:
My understanding here was: maxpatchpublishdate can contain value set from inclusions. Default Patching sends in healtstoreId, that's the reason behind using healthstoreId here to compute posix time.

Reply:
Maxpatchpublishdate is level of abstraction visible outside. Everything below that is supposed to be invisible to package managers for SDP (where the data comes from, what it contains before being surfaced as maxpatchpublishdate, etc.).

Reply:
Specifically look at how this is implemented in Apt - APM has zero knowledge of where maxpatchpublishdate is coming from at any point. There's a value there or there isn't - and it accommodates both and behaves accordingly. Healthstore id is completely an unknown entity to the specialized APM class. It means nothing to the level of knowledge it needs to have.

Reply:
@kjohn-msft please review the latest changes